tests/test_x509.py

```diff
@@ -1458,17 +1461,16 @@ def test_build_cert_printable_string_country_name(self, backend):
 
         cert = builder.sign(issuer_private_key, hashes.SHA256(), backend)
 
-        parsed, _ = decoder.decode(
-            cert.public_bytes(serialization.Encoding.DER),
-            asn1Spec=rfc2459.Certificate()
-        )
-        tbs_cert = parsed.getComponentByName('tbsCertificate')
-        subject = tbs_cert.getComponentByName('subject')
-        issuer = tbs_cert.getComponentByName('issuer')
-        # \x13 is printable string. The first byte of the value of the
-        # node corresponds to the ASN.1 string type.
-        assert subject[0][0][0][1][0] == b"\x13"[0]
-        assert issuer[0][0][0][1][0] == b"\x13"[0]
+        parsed = Certificate.load(
+            cert.public_bytes(serialization.Encoding.DER))
+        tbs_cert = parsed['tbs_certificate']
+        subject = tbs_cert['subject'].native
+        issuer = tbs_cert['issuer'].native
+
+        assert subject['country_name'] == country_name
+        assert subject['state_or_province_name'] == state_or_province_name
+        assert issuer['country_name'] == country_name
+        assert issuer['state_or_province_name'] == state_or_province_name
```

Comment:
This changed the assertion we were making -- this test is to specifically ensure that the ASN.1 encoding is correct, it's not concerned with the value.